### ISSUE TITLE
read_lines problem in a long running process

### ISSUE THREAD

**Issue by kensanata** [Leont/file-slurp-sane#10]
I recently had to remove `read_lines` from a little server I am developing ([Phoebe](https://github.com/kensanata/phoebe/), [commit](https://github.com/kensanata/phoebe/commit/b3016b18b0e66a986bac40e4192492df7cec2791)). It used `read_lines` to read a list of "pages" from an "index" file. When I started or restarted the server, everything worked as intended, the index showed one item per page. After a few hours, the behaviour changed: `read_lines` returned a single item containing all the pages concatenated with newlines.

Normal operations with an index containing three files:

1. A
2. B
3. C

After a few hours, with the same index:

1. A\nB\nC

And when I restart the process, it is back to three items.

Sadly, I have no idea how this is possible. I never pass `$encoding`, `$crlf`, or `$skip_chomp`, and `read_text` also calls `_text_layers` just like `read_lines`. I'm just confused about the situation right now and don't know how I should debug or log the issue.

**Comment by Leont** [comment]
Could it be something elsewhere in your stack sets `$/`? I'm currently not protecting against that, and I probably should.

**Comment by kensanata** [comment]
Interesting idea! Thanks. I will investigate…

```text
lib/App/Phoebe/Gopher.pm:      local $/ = undef; # slurp
lib/App/Phoebe/Spartan.pm:      local $/ = undef; # slurp
script/spartan:  local $/ = undef;
t/cert.pl:  local $/ = undef;
t/oddmuse-wiki.pl:    local $/ = undef; # Read complete files
t/oddmuse-wiki.pl:    local $/ = undef;         # Read complete files
README.md:      undef $/; # slurp
lib/App/Phoebe/Oracle.pm:      undef $/; # slurp
script/titan:undef $/;
t/00_tls_check.t:  undef $/; # slurp
t/Gopher.t:  undef $/; # slurp
t/Iapetus.t:  undef $/; # slurp
t/Spartan.t:  undef $/; # slurp
t/oddmuse-gopher.t:  undef $/; # slurp
```

**Comment by kensanata** [comment]
I think I found the culprit. In the listing above there is a setting of `$/` that isn't local in `Oracle.pm` – at first I discounted that match because it only showed up in the documentation as a code example, but of course it turns out that I used that code in my `config` file in production in a piece of code that only rarely got called (every few days). Without further testing I'd say that explains the behaviour and I'll close the issue.

Thanks for that pointer, and for `File::Slurper`.
